llms/googleai/googleai_llm.go

```diff
@@ -31,11 +32,18 @@ const (
 	SAFETY    = "safety"
 )
 
-// New creates a new GoogleAI struct.
-func New(ctx context.Context, options ...option.ClientOption) (*GoogleAI, error) {
-	gi := &GoogleAI{}
+// NewGoogleAI creates a new GoogleAI struct.
+func NewGoogleAI(ctx context.Context, opts ...Option) (*GoogleAI, error) {
+	clientOptions := defaultOptions()
+	for _, opt := range opts {
+		opt(&clientOptions)
+	}
+
+	gi := &GoogleAI{
+		opts: clientOptions,
+	}
 
-	client, err := genai.NewClient(ctx, options...)
+	client, err := genai.NewClient(ctx, option.WithAPIKey(clientOptions.apiKey))
```

Comment:
I know there was some discussion regarding ADC on the issue -- does this preclude falling back to ADC?

Reply:
Nope, that will be a separate client hidden behind the abstraction in langchaingo -- see https://github.com/tmc/langchaingo/issues/410#issuecomment-1879717569